src/config_brain_memory.js

```diff
@@ -47,7 +47,10 @@ brain.prepareMemory = function() {
   for (let name in Memory.creeps) {
     if (!Game.creeps[name]) {
       let role = Memory.creeps[name].role;
-      let roleStat = Memory.stats[userName].roles[role];
+      let roleStat;
+      if (!Memory.stats[userName]) {
```

Comment:
if Memory.stats[username] is undefined then roleStat = Memory.stats[userName].roles[role]; is undefined too